### ISSUE TITLE
Can APM monitor server-only code?

### ISSUE THREAD

**Issue by dandv** [arunoda/meteor-apm-client#35]
I'm back to using APM after a few months. Can it monitor apps where there is no client? I have a backend that aggregates RSS feeds and it routinely jumps to 100% CPU. Would be great to see what causes that.

The app id is `L73TPeNfBsevKvJ9M` and I've launched it a few hours ago, but there's no data. The frontend (`6kpgrQKaEf52bsb8z`) was launched the same way and data comes in fine.

**Comment by arunoda** [comment]
What do you mean by backend, is it a meteor app?

On Tuesday, May 13, 2014, Dan Dascalescu notifications@github.com wrote:

> I'm back to using APM after a few months. Can it monitor apps where there
> is no client? I have a backend that aggregates RSS feeds and it routinely
> jumps to 100% CPU. Would be great to see what causes that.
> 
> The app id is L73TPeNfBsevKvJ9M and I've launched it a few hours ago, but
> there's no data. The frontend was launched the same way (6kpgrQKaEf52bsb8z)
> and data comes in fine.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-apm-client/issues/35
> .

## 

---

Arunoda Susiripala
I curate Meteor Weekly - Check it
out!http://meteorhacks.com/meteor-weekly/?utm_source=email-footer&utm_medium=email&utm_campaign=meteorweekly

**Comment by dandv** [comment]
Yes, the backend is a Meteor app with only server code, which does RSS feed aggregation and other stuff. (I'll be adding an admin interface later, using Houston).

**Comment by arunoda** [comment]
Okay cool.

If it does not have methods and pubsub, you will not get metrics. We are
adding System metrics very soon.
They will work without the methods and pubsub.

Am I correct?

---

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
 http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda

**Comment by dandv** [comment]
The server code has methods, but no pubsub. It does use a bunch of collections.

**Comment by arunoda** [comment]
I check our DB and it didn't received any data from this app too.

Can you check following message displayed on the log?

``` js
APM: completed instrumenting the app
APM: successfully authenticated
```

If you send me a minimal repo, I could dig into more and find a fix quickly. Thanks.

**Comment by dandv** [comment]
Correct, I do see the two messages.

This backend code has no `Meteor.methods` or publications called by the client, which explains why there's nothing to monitor yet. (The backend basically uses PowerQueue to parse a bunch of RSS feeds, process the data, and store it in Mongo.)

**Comment by arunoda** [comment]
We will launch cpu and memory metrics with the APM release. That's only
you'll be able to track until the next big change.

On Monday, May 19, 2014, Dan Dascalescu notifications@github.com wrote:

> Correct, I do see the two messages.
> 
> This backend code has no Meteor.methods or publications called by the
> client, which explains why there's nothing to monitor yet. (The backend
> basically uses PowerQueue to parse a bunch of RSS feeds, process the data,
> and store it in Mongo.)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-apm-client/issues/35#issuecomment-43510952
> .

## 

---

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda
